src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,503 @@
+    hbm
+};
+
+// Presumably a struct in anticipation of more data
+struct config
+{
+  std::string host;
+
+  void write(std::ostream& ostr)
+  {
+    ostr << "host=" << host << std::endl;
+  }
+};
+
+SubCmdConfigure::SubCmdConfigure(bool _isHidden, bool _isDepricated, bool _isPreliminary)
+    : SubCmd("configure",
+             "Advanced options for configuring a device")
+{
+  const std::string long_description = "Advanced options for configuring a device";
+  setLongDescription(long_description);
+  setExampleSyntax("");
+  setIsHidden(_isHidden);
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+// So far, we only support the following configs, eg.
+// [Device]
+// mailbox_channel_disable = 0x20
+// mailbox_channel_switch = 0
+// xclbin_change = 1
+// cahce_xclbin = 0
+// we may support in the future, like,
+// [Daemon]
+// host_ip = x.x.x.x
+static void load_config(const std::shared_ptr<xrt_core::device>& _dev, const std::string path)
+{
+  boost::property_tree::ptree pt_root;
+  boost::property_tree::ini_parser::read_ini(path, pt_root);
+  static boost::property_tree::ptree empty_tree;
+
+  const boost::property_tree::ptree pt_device = pt_root.get_child("Device", empty_tree);
+
+  if (pt_device.empty())
+    throw std::runtime_error(boost::str(boost::format("No [Device] section in the config file. Config File: %s") % path));
+
+  for (auto& key : pt_device) {
+    if (!key.first.compare("mailbox_channel_disable")) {
+      xrt_core::device_update<xrt_core::query::config_mailbox_channel_disable>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("mailbox_channel_switch")) {
+      xrt_core::device_update<xrt_core::query::config_mailbox_channel_switch>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("xclbin_change")) {
+      xrt_core::device_update<xrt_core::query::config_xclbin_change>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("cache_xclbin")) {
+      xrt_core::device_update<xrt_core::query::cache_xclbin>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("scaling_enabled")) {
+      xrt_core::device_update<xrt_core::query::xmc_scaling_enabled>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("scaling_power_override")) {
+      xrt_core::device_update<xrt_core::query::xmc_scaling_power_override>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    if (!key.first.compare("scaling_temp_override")) {
+      xrt_core::device_update<xrt_core::query::xmc_scaling_temp_override>(_dev.get(), key.second.get_value<std::string>());
+      continue;
+    }
+    throw std::runtime_error(boost::str(boost::format("'%s' is not a supported config entry") % key.first));
+  }
+}
+
+static config
+get_daemon_conf()
+{
+  config cfg;
```

Comment:
config is a structure and should be initialized.
```suggestion
  config cfg = {0};
```

Reply:
I initialized it like so:
config cfg = {.host = ""};
Hope that is alright